libraries/disp3D/engine/model/items/sensordata/cpusensordatatreeitem.h

```diff
@@ -0,0 +1,268 @@
+*
+*/
+
+#ifndef DISP3DLIB_CPUSENSORDATATREEITEM_H
+#define DISP3DLIB_CPUSENSORDATATREEITEM_H
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include "../../../../disp3D_global.h"
+#include "sensordatatreeitem.h"
+
+//*************************************************************************************************************
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QSharedPointer>
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// Eigen INCLUDES
+//=============================================================================================================
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// FORWARD DECLARATIONS
+//=============================================================================================================
+
+namespace FIFFLIB{
+    class FiffEvoked;
+}
+
+namespace MNELIB{
+    class MNEBemSurface;
+}
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE NAMESPACE DISP3DLIB
+//=============================================================================================================
+
+namespace DISP3DLIB {
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DISP3DLIB FORWARD DECLARATIONS
+//=============================================================================================================
+
+class RtSensorDataWorker;
+
+//=============================================================================================================
+/**
+* This item allows on-the-fly changes to parameters of visualization. It integrates the features provided in
+* GeometryInfo and and uses the cpu for interpolation.
+*
+* @brief This item integrates GeometryInfo and  Interpolation into Disp3D structure.
+*/
+class DISP3DSHARED_EXPORT CpuSensorDataTreeItem : public SensorDataTreeItem
```

Comment:
almost all functions are overwritten by this new item class. Does it make sense to Keep SensorDataTreeItem ? Was there a Special reason why you inherit from SensorDataTreeItem ?